tmt/steps/execute/internal.py

```diff
@@ -631,20 +631,33 @@ def _save_process(
         # overwrite it.
         self.write(invocation.path / TEST_OUTPUT_FILENAME, stdout or '', mode='a', level=3)
 
+        def pull_from_guest() -> None:
+            if not invocation.is_guest_healthy:
+                return
+
+            try:
+                guest.pull(
+                    source=invocation.path,
+                    extend_options=[
+                        *test.test_framework.get_pull_options(invocation, logger),
+                        '--exclude',
+                        str(invocation.path / TEST_OUTPUT_FILENAME),
+                    ],
+                )
+
+                # Fetch plan data content as well in order to prevent
+                # losing logs if the guest becomes later unresponsive.
+                guest.pull(source=self.step.plan.data_directory)
+
+            # Handle failing to pull test artifacts after guest becoming
+            # unresponsive. If not handled test would stay in 'pending' state.
+            # See issue https://github.com/teemtee/tmt/issues/3647.
+            except tmt.utils.RunError:
+                pass
```

Comment:
-1 from me: if this fails, no one will ever know; at least a warning would be nice. It's not handling an error, it's silencing without saying so.

Reply:
@happz there is already a reasonable failure messaged printed, see:

https://github.com/teemtee/tmt/blob/169e32dd77686d935fa3f74f3fbb53a12d8b9d4a/tmt/steps/provision/__init__.py#L2490

I did not find it useful to inform the user more.

Reply:
Please add a large TODO here, I will try to improve this because I consider it a risky approach. The fact that I need to go deeper into the traceback to find out there was indeed an error message is a clear sign this should be better. I guess it is not your problem here and now, but this should be improved.

> I did not find it useful to inform the user more.

I think this reveals the issue: your code encounters an exception, and we assume the called code reported the error. If anyone removes that `fail` from `pull`, this will break down, and nothing will report the problem because those two places are detached.

Reply:
@happz added, please re-review